### ISSUE TITLE
Error creating load balancer

### ISSUE THREAD

**Issue by 7wonders** [rancher/rancher#10429]
**Rancher versions:**
rancher/server: v1.6.11

**Infrastructure Stack versions:**
kubernetes: rancher/k8s:v1.8.3-rancher2

**Operating system and kernel:**
RancherOS v1.1.0 (4.9.45)

**Type/provider of hosts:** 
AWS

**Setup details:**
Full HA AWS external DB

**Environment Template:**
Kubernetes (aws cloud provider)

**Steps to Reproduce:**
Follow the steps in the docs to add an ELB load balancer service:
https://rancher.com/docs/rancher/v1.6/en/kubernetes/providers/#elastic-load-balancer-elb-as-a-kubernetes-service

**Results:**
Error creating load balancer (will retry): Failed to ensure load balancer for service default/nginx-lb: Multiple tagged security groups found for instance i-0df*****; ensure only the k8s security group is tagged

Found this: https://github.com/kubernetes/kubernetes/issues/26787

**Comment by 7wonders** [comment]
I will add that it creates the ELB with the tag:

kubernetes.io/service-name | default/nginx-lb

Is rancher/k8s supposed to add some tags to the hosts automatically?

**Comment by 7wonders** [comment]
To report back in, this happens if choosing multiple security groups during host set up. Recreating the hosts with a single security group fixed the problem. I will leave the issue open as I still think its a bug.

**Comment by truedat101** [comment]
Wow this is brutal.  I spent a bunch of time trying to figure out why the ELB kept failing with the error results in the problem description.  The description of the problem reported by k8s in terms of service failure  is confusing.  "Multiple tagged security groups found for instance i-SOMEID" caused me to think I had a problem somewhere in my AWS tags attached to the SG for k8s, where the problem was as you described, that I had accidentally set up the instance with multiple SGs during setup (just this one host for some reason), and k8s can't seem to clean up the facts even if I remove one of the security groups from this host on the AWS side.  It's not clear that this is a rancher problem necessarily, but something down in k8s.  I guess a tag attribute gets attached to something down in the plumbing?  New to k8s, so I got stuck on this  for hours.

**Comment by truedat101** [comment]
I believe one solution is to improve documentation in this area.  It is really nice to be able to use rancher to add hosts, but the caution might need to existing in the k8s configuration around adding hosts, if indeed the problem is having multiple SGs assigned will goof up ELB creation.

**Comment by gigi-at-zymergen** [comment]
Thanks for pointing out that you had to actually recreate the hosts rather than just clean up the security groups @truedat101 . That save me some time when I found this issue.

**Comment by stonefury** [comment]
Do I need to remove only the workers and then re-add them using the custom add host?  Didn't seem to work. Or do I need to re-add every single host in the k8s cluster?

**Comment by 7wonders** [comment]
@stonefury you have to add the hosts with a single security group during host set up. First fix a security group that covers your needs, then just do a clone on your existing hosts but use only the single security group. When the clone is up, drop the broken instance. Rinse and repeat until done. Should work.

**Comment by stonefury** [comment]
@7wonders really appreciate your advice.  I have it working.  I created a brand new cluster with EC2 instances with only 1 sg, and that worked fine.

My existing cluster, I created new EC2 instances with one security group and added them and that worked too.  Actually I had rancher provision the EC2 instance for me and generate it's own security group, and that worked. I could create the elb after that.
